### ISSUE TITLE
Specifying channel for package in yaml file does not work for <domain:port> channels

### ISSUE THREAD

**Issue by zbs** [conda/conda#10407]
Per [this issue](https://github.com/conda/conda/issues/5693), Conda supports specifying per-dependency channels in the yaml file; however, a yaml file such as this:

```
name: bar
channels:
  - http://foo_domain:port
dependencies:
  - python=3.7
  - http://foo_domain:port::package=1.1.0
prefix: /foo
```

yields the following error when running `conda env create -f config.yml -p ./fooenv`:

```
Collecting package metadata (repodata.json): done
Solving environment: failed

ResolvePackageNotFound: 
  - <unknown>::package=1.0.0
```

By contrast, the following works:
`conda install -c http://foo_domain:port package=1.1.0`

**Comment by cjmartian** [comment]
Hey @zbs, 

Thanks for pointing this out to us. I'm going to get a PR opened for this.

**Comment by chebee7i** [comment]
`file://` seems to be similarly affected.

**Comment by cjmartian** [comment]
https://github.com/conda/conda/pull/10417

**Comment by cjmartian** [comment]
I'm going to look into `file://` as well

**Comment by chebee7i** [comment]
Any update on `file://` support?

I tried with `conda 4.10.3` and that did not seem to work---it was released on `4.9.3` from what I can see.

My yaml has:
```console
 channels:
   - file:///jcs/public/conda/jcs-pkgs
 dependencies:
   - file:///jcs/public/conda/jcs-pkgs::mypackage
```

But when installing, we got:
```console
UnavailableInvalidChannel: The channel is not accessible or is invalid.
  channel name: mychannel
  channel url: https://conda.anaconda.org/mychannel
  error code: 404

**Comment by chebee7i** [comment]
@cjmartian any chance we can get this fixed?   It is still not functional for `file:///`
